src/routers/check.details.router.ts

```diff
@@ -1,13 +1,23 @@
-import { Request, Response, Router, NextFunction } from "express";
+import { NextFunction, Request, Response, Router } from "express";
 import { CheckDetailsHandler } from "./handlers/check_details";
-
+import { wrapAsyncHandlerWithErrorHandler } from "../utils/async.handler";
 const router: Router = Router();
 
-router.get("/", (req: Request, res: Response, _next: NextFunction) => {
+router.get("/", wrapAsyncHandlerWithErrorHandler(async (req: Request, res: Response) => {
     const handler = new CheckDetailsHandler();
-    const { templatePath, viewData } = handler.execute(req, res);
+    const { templatePath, viewData } = handler.executeGet(req, res);
     res.render(templatePath, viewData);
-});
+}));
+
+router.post("/", wrapAsyncHandlerWithErrorHandler(async (req: Request, res: Response, next: NextFunction) => {
+    const handler = new CheckDetailsHandler();
+    const resp = await handler.executePost(req, res);
```

Comment:
I think give this variable a more descriptive name as it is almost identical to res

Reply:
I've renamed it to `submitDetailsResult`. Hopefully that is a bit better.